demo/src/components/CurrentSlide.tsx

```diff
@@ -12,7 +12,7 @@ export const CurrentSlide = ({ slides }: Props) => {
   const { self } = useMembers();
   const slide = parseInt(self?.location?.slide || '', 10) || 0;
 
-  useTrackCursor(containerRef);
+  useTrackCursor(containerRef, self?.connectionId);
```

Comment:
Would you mind explaining why you're checking `self?.connectionId` instead of just `self`? The [documentation for `getSelf()`](https://github.com/ably/spaces/blob/916404e0a9b3f8328e91fa46e8bdcb76cc5f4020/docs/class-definitions.md?plain=1#L247), which is what we use to populate `self` here, says:

> Will resolve to `null` if the client hasn't entered the space yet.

Reply:
Of course. React cares about referential quality in its hooks dependencies and props. It's quite likely that `self` will be a different object but with the same values, so I use the `connectionId` which will be either `undefined` or a `string` which should mean renders are not triggered for the same data (or subscriptions tore down and resubscribed without purpose).

Reply:
Thanks for the explanation. I think this is hitting against the boundaries of my React knowledge so I'll take your word on this one 🙂